app/controllers/application_controller.rb

```diff
@@ -1,12 +1,20 @@
 class ApplicationController < ActionController::Base
+  before_filter :configure_permitted_parameters, if: :devise_controller?
+  before_action :set_locale
```

Comment:
why do you need this action before each request. If you wanted to provide multiple languages feature set to I18n.locale  
some default locale(e.g. Ukrainian) within `config file`, and change it on click some language button.